crates/eips/src/eip1898.rs

```diff
@@ -232,6 +232,85 @@ impl fmt::Debug for BlockNumberOrTag {
     }
 }
 
+/// A module that deserializes either a BlockNumberOrTag, or a simple number.
+pub mod lenient_block_number_or_tag {
```

Comment:
can we not copy paste the existing module and then add the 

```
pub struct LenientBlockNumberOrTag(BlockNumberOrTag);
```

as a new public enum?
this seems easier than doing this

we also want some additional test cases for this